controllers/nutanixmachine_controller.go

```diff
@@ -88,9 +87,51 @@ func (r *NutanixMachineReconciler) SetupWithManager(ctx context.Context, mgr ctr
 				capiutil.MachineToInfrastructureMapFunc(
 					infrav1.GroupVersion.WithKind("NutanixMachine"))),
 		).
+		Watches(
+			&source.Kind{Type: &infrav1.NutanixCluster{}},
+			handler.EnqueueRequestsFromMapFunc(r.mapNutanixClusterToNutanixMachines(ctx)),
+		).
 		Complete(r)
 }
 
+func (r *NutanixMachineReconciler) mapNutanixClusterToNutanixMachines(ctx context.Context) handler.MapFunc {
+	return func(o client.Object) []ctrl.Request {
+		nutanixCluster, ok := o.(*infrav1.NutanixCluster)
+		if !ok {
+			klog.Errorf("Expected a NutanixCluster object in mapNutanixClusterToNutanixMachines but was %T", o)
+			return nil
+		}
+		logPrefix := fmt.Sprintf("NutanixCluster[namespace: %s, name: %s]", nutanixCluster.Namespace, nutanixCluster.Name)
+
+		cluster, err := util.GetOwnerCluster(context.TODO(), r.Client, nutanixCluster.ObjectMeta)
```

Comment:
we should be passing the context along, not creating a new one

Reply:
Changed